src/dmd/expressionsem.d

```diff
@@ -6833,8 +6851,8 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                             }
                         }
 
-                        Expression einit;
-                        einit = new BlitExp(exp.loc, e1x, e1x.type.defaultInit(exp.loc));
+                        Expression einit = getInitExp(sd, exp.loc, sc, t1);
+                        einit = new BlitExp(exp.loc, e1x, einit);
```

Comment:
Note to other reviewers: it's fine that this PR replaces `e1x.type` with  `t1` because here `e1x` is `exp.e1` and `t1` is `exp.e1.type.toBasetype()`.

Reply:
Note that `einit` is assigned twice. A better method is to assign it only once.

Reply:
Fixed.